compiler/rustc_pattern_analysis/src/rustc/print.rs

```diff
@@ -60,7 +81,18 @@ pub(crate) fn write_struct_like<'tcx>(
             {
                 variant.name.to_string()
             } else {
-                format!("{}::{}", tcx.def_path_str(adt_def.did()), variant.name)
+                let enum_and_variant = if let Some(scrut) = scrut
+                    && erase_path_if_local(tcx, adt_def, scrut)
+                {
+                    ty::print::with_forced_trimmed_paths!(format!(
+                        "{}::{}",
+                        tcx.def_path_str(adt_def.did()),
+                        variant.name
+                    ))
+                } else {
+                    format!("{}::{}", tcx.def_path_str(adt_def.did()), variant.name)
```

Comment:
nit: this line is used identically in both arms of the `if`, you can remove this duplication